GymApp/GYM.IntegrationTests/IntegrationTests/CouchesControllerIntegrationTests.cs

```diff
@@ -0,0 +1,162 @@
+﻿using AutoFixture;
+using GYM.API.Models;
+using GYM.DAL.EF;
+using GYM.DAL.Entities;
+using Microsoft.AspNetCore.Mvc.Testing;
+using Microsoft.AspNetCore.TestHost;
+using Microsoft.EntityFrameworkCore;
+using Microsoft.Extensions.DependencyInjection;
+using Shouldly;
+using System.Net;
+using System.Net.Http.Json;
+
+namespace GYM.API.IntegrationTests.IntegrationTests
+{
+    public class CouchesControllerIntegrationTests
+    {
+        private readonly HttpClient _client;
+        private readonly GymAppDbContext _dbContext;
+        private readonly Fixture _fixture;
+        private const string RouteWithoutId = "api/Couches";
+        private const string RouteWithId = "api/Couches/";
+
+        public CouchesControllerIntegrationTests()
+        {
+            _fixture = new Fixture();
+
+            var webHost = new WebApplicationFactory<Program>().WithWebHostBuilder(builder =>
+            {
+                builder.ConfigureTestServices(services =>
+                {
+                    var dbContextDescriptor = services.SingleOrDefault(d =>
+                        d.ServiceType == typeof(DbContextOptions<GymAppDbContext>));
+
+                    services.Remove(dbContextDescriptor!);
+
+                    services.AddDbContext<GymAppDbContext>(options =>
+                    {
+                        options.UseInMemoryDatabase("InMemoryGymAppTest");
+                    });
+                });
+            });
+
+            _client = webHost.CreateClient();
+            _dbContext = webHost.Services.CreateScope().ServiceProvider.GetService<GymAppDbContext>()!;
+            _dbContext.CouchEntities.AddRange(GetCouchesEntityForTest());
+            _dbContext.SaveChanges();
+        }
+
+        [Fact]
+        public async Task GetCouches_HasNotData_ReturnsStatusOkAndAllCouches()
+        {
+            //Arrange
+            var couch = _dbContext.CouchEntities.LastOrDefault();
+
+            //Act
+            var response = await _client.GetAsync(RouteWithoutId);
+            var responseString = await response.Content.ReadAsStringAsync();
```

Comment:
You also can use "await response.Content.ReadFromJsonAsync<CouchEntities>() " for get a list of all coaches and than find coach who has that name